File: src/other/median-of-two-sorted-arrays/median-of-two-sorted-arrays.ts
function findMedianSortedArrays(nums1: number[], nums2: number[]): number {
    // Not the fastest solution, but it will work. Merge both arrays,
    // re-sort, and return the median.
    const merged = [...nums1, ...nums2].sort((n1, n2) => n1 - n2);
    return median(merged);
};

const median = (nums: number[]): number => {
    // Handle zero-length case
    if (nums.length === 0) {
        return 0;
    }

    // Handle even length case
    if (nums.length % 2 === 0) {
        const lower = (nums.length / 2) - 1;
        const upper = (nums.length / 2);
        return (nums[lower] + nums[upper]) / 2;
    }

    // Handle odd length case
    const middle = Math.floor(nums.length / 2);
    return nums[middle];
}

File: src/other/median-of-two-sorted-arrays/median-of-two-sorted-arrays.rs
impl Solution {
    pub fn find_median_sorted_arrays(nums1: Vec<i32>, nums2: Vec<i32>) -> f64 {
        let mut merged = nums1.clone();
        merged.append(&mut nums2.clone());
        merged.sort();
        let totalNums = merged.len();
        match totalNums % 2
        {
            0 =>
            {
                let lowerIdx = (totalNums / 2) - 1;
                let upperIdx = lowerIdx + 1;
                let lowerMedian = merged.get(lowerIdx).unwrap();
                let upperMedian = merged.get(upperIdx).unwrap();
                (*lowerMedian as f64 +
                 *upperMedian as f64) / 2f64
            }
            // Will always be 1, but compiler does not know that
            _ =>
            {
                let middle = totalNums / 2;
                let median = merged.get(middle).unwrap();
                *median as f64
            }
        }
    }
}

